### PR TITLE
Refactored data clumps with the help of LLMs (research project)

### PR DESCRIPTION
**Pull request by compf**
Hello maintainers,

I am conducting a master thesis project focused on enhancing code quality through automated refactoring of data clumps, assisted by Large Language Models (LLMs).

<details>
  <summary>Data clump definition</summary>
  
A data clump exists if
1. two methods (in the same or in different classes) have at least 3 common parameters and one of those methods does not override the other,  or 
2. At least three fields in a class are common with the parameters of a method (in the same or in a different class), or
3. Two different classes have at least three common fields
  
See also the following UML diagram as an example
Example data clump
</details>

I believe these refactoring can contribute to the project by reducing complexity and enhancing readability of your source code.

Pursuant to the EU AI Act, I fully disclose the use of LLMs in generating these refactorings, emphasizing that all changes have undergone human review for quality assurance. 

Even if you decide not to integrate my changes to your codebase (which is perfectly fine), I ask you to fill out a feedback survey, which will be scientifically evaluated to determine the acceptance of AI-supported refactorings. You can find the feedback survey under https://campus.lamapoll.de/Data-clump-refactoring/en

Thank you for considering my contribution. I look forward to your feedback. If you have any other questions or comments, feel free to write a comment, or email me under tschoemaker@uni-osnabrueck.de .

Best regards,
Timo Schoemaker
Department of Computer Science
University of Osnabrück

### CODE REVIEW

File: src/net/sourceforge/plantuml/klimt/drawing/eps/DriverTextEps.java [hunk 1]
```diff
@@ -34,13 +34,13 @@
  */
 package net.sourceforge.plantuml.klimt.drawing.eps;
 
-import java.awt.Color;
-import java.awt.Shape;
+import java.awt.*;
```
Comment (The-Lum) [inline src/net/sourceforge/plantuml/klimt/drawing/eps/DriverTextEps.java:37]: Here is a minor remark:
- Be careful with global imports `import xxx.*;`

[FYI @arnaudroques]

Comment (compf) [reply src/net/sourceforge/plantuml/klimt/drawing/eps/DriverTextEps.java:37]: > Here is a minor remark:
> 
>     * Be careful with global imports `import xxx.*;`
> 
> 
> [FYI @arnaudroques]

Thank you for the feedback :)

**Comment by arnaudroques** [comment]
Thanks for your contribution! Use of LLM is indeed very useful.
The creation of `RectangleCoordinates` is a good point. However, I would probably set the class as immutable (that is, without any setter).
We are not going to integrate your change asis, however, we will probably go for your suggestion (that is, an `RectangleCoordinates` immutable class), maybe with a shorter name for the class.

Your MR will be very useful to see all impact of this change on your code.
Thanks again!

**Comment by compf** [comment]
> Thanks for your contribution! Use of LLM is indeed very useful. The creation of `RectangleCoordinates` is a good point. However, I would probably set the class as immutable (that is, without any setter). We are not going to integrate your change asis, however, we will probably go for your suggestion (that is, an `RectangleCoordinates` immutable class), maybe with a shorter name for the class.
> 
> Your MR will be very useful to see all impact of this change on your code. Thanks again!

Thank you very much for the feedback :)

**Pull request closed without merging**